src/book/02-system/04-hardware/04-dynamixel-protocol.mdx

```diff
@@ -0,0 +1,388 @@
+
+The goal of this guide is to give an overview of the Dynamixel communication. It is not meant to be an exhaustive list of what is already on the Robotis [manual](https://emanual.robotis.com/docs/en/dxl/protocol2/) but to give further insight into things left out of said manual, mostly the low-level electrical side, as well as any idiosyncracities. Said manual should be read after the reader has read this guide.
+
+The target-audience is somewhat broad and can be anywhere between
+
+- a team-member who is experienced with the Dynamixel system but wants to cross-reference something,
+- a new electrical, computer, or mechatronics engineering student who has had some experience with some of these concepts such as UART in ELEC2720 but may want to refresh some of the basics and needs to learn some new things such as RS485 and the Dynamixel protocol,
+- and a computer-science student who has some experience with telecommunications such as Ethernet but may have no experience at all with UART and RS485.
+
+<Alert type="warn">
+
+Assumed knowledge is a basic understanding of electrical concepts like voltage and maybe some HSC physics such as magnetic fields as well as a basic intuition for electronic devices in daily life.
+
+</Alert>
+
+## Universal Asynchronous Receiver-Transmitter
+
+A universal asynchronous receiver-transmitter (UART) is an electrical device that sends data at an agreed upon bitrate or baudrate. Unlike SPI and I2C where there is one extra signal for the clock, often just called the serial clock, both devices on either end have to know the common baudrate beforehand. Thus, it is common to have to set a baudrate at the very beginning externally either by rewriting firmware or configuring a peripheral.
+
+<Alert type="info">
+
+There is also a universal synchronous and asynchronous receiver-transmitter (USART) which is a device that can handle both synchronous and asynchronous communication. Many microcontrollers would have both USARTs and UARTs, the former of which can more of less be used as the latter. However, we will only talk about the asynchronous side in which a UART henceforth can easily be applied for a USART as well.
+
+</Alert>
+
+A UART has two pins, an input called RX, and an output called TX.
+
+![Two abstract UARTs](./images/dynamixel_protocol/pure_uart.drawio.png 'Two abstract UARTs have their RX-pins and TX-pins connected each way.')
+
+The term UART can also refer to the way how data is framed. If one will, it can almost be informally thought of as a very low-level and very basic protocol. Here, UART can refer to the special bits that are needed to frame data, all of which need to be agreed upon between both devices on either end.
```

Comment:
```suggestion
The term UART can also refer to the way data is framed. It can be informally thought of as a very low-level and very basic protocol. Here, UART can refer to the special bits that are needed to frame data, all of which need to be agreed upon between both devices on either end.
```